### DISCUSSION TITLE
TypeError: x.replace is not a function after migration from 0.X to 1.X.X

### DISCUSSION THREAD

**Discussion by mentasuave01** [wevm/wagmi#2972] [Q&A]
Same code I didn't change a thing. Only a few things in wagconfig to make the migration but the rest should work.
I have this error using: 
```js
const { config, error } = usePrepareContractWrite({...data})
```
error:
`TypeError: x.replace is not a function
    at eval (webpack-internal:///(:3080/app-client)/./node_modules/.pnpm/viem@1.7.0_typescript@5.0.4/node_modules/viem/dist/esm/utils/data/concat.js:26:51)
    at Array.reduce (<anonymous>)
    at concatHex (webpack-internal:....)`

**Comment by mentasuave01** [comment]
Changing view dep from view 1.x.x to "viem": "~0.3.36" Solves the issue but this needs further research 
@tmm

**Reply by mentasuave01** [reply]
up

**Reply by mentasuave01** [reply]
up 2

**Comment by hardingjam** [comment]
Are you passing an empty array as an arg? If so, I was able to fix this by passing "0x0000000000000000000000000000000000000000000000000000000000000000" explicitly.

For me, this issue arose after migrating to wagmi 2.x from 1.x
I couldn't find any mention of a change in the docs, but i think it was a viem issue
